### ISSUE TITLE
YAML::Store does not retrieve UTF8 values correctly in read-only mode

### ISSUE THREAD

**Issue by duritong** [jruby/jruby#4770]
While investigating further problems on 9.1.7.0 related to #4767 I came accross a weird behavior of `YAML::Store` which does not print out correct unicode characters if the transaction is in readonly.

This is a small script that shows the problem

```
$ ruby --version
jruby 9.1.7.0 (2.3.1) 2017-01-11 68056ae OpenJDK 64-Bit Server VM 25.141-b16 on 1.8.0_141-b16 +jit [linux-x86_64]
$ cat pstore_test.rb 
require 'yaml/store'

puts 'PStore'
store = PStore.new('test.pstore')
store.transaction{ store['test'] = 'Tütü' }
print 'Correct: '
puts store.transaction{ store['test'] }
print 'Correct: '
puts store.transaction(true){ store['test'] }

puts 'YAML'
store = YAML::Store.new('test.yaml')
store.transaction{ store['test'] = 'Tütü' }
print 'Correct: '
puts store.transaction{ store['test'] }
print 'Incorrect: '
puts store.transaction(true){ store['test'] }
$ rm test.*
$ ruby pstore_test.rb 
PStore
Correct: Tütü
Correct: Tütü
YAML
Correct: Tütü
Incorrect: TÃ¼tÃ¼
```

**Comment by duritong** [comment]
I added a testcase in #4771 that shows the problem and hopefully will avoid future regressions.

**Comment by enebo** [comment]
Reduced this a bit:

```text
jruby -ryaml -e 'p YAML.load(File.open("test.yaml", {encoding: Encoding::ASCII_8BIT}))["test"]'
```

If I change this to UTF-8 it works so I am guessing YAML as a library might be capable of negotiating ASCII_8BIT to valid UTF-8 and our version doesn't?

**Comment by enebo** [comment]
I submitted a PR for this to psych but I will work around it locally in yaml/store until that new version is out so you can see this fixed in 9.1.13.0.

**Comment by headius** [comment]
Link to #4784.
